homeassistant/components/media_player/samsungtv.py

```diff
@@ -155,18 +154,17 @@ def send_key(self, key):
             _LOGGER.info("TV is powering off, not sending command: %s", key)
             return
         try:
-            self.get_remote().control(key)
+            with self.get_remote() as remote:
```

Comment:
Creating and closing a new connection for every key seems expensive. Why not retry the command with a new connection if we get a `ConnectionClosed` ?

Reply:
Would it be possible for me to IM with you at your preferred time so I can clarify and avoid lengthy back and forth?

I can see you say that for (web)socket connections to it's a pattern to keep connection open. My question has been why that's the case for components using (web)socket but not for those that use HTTP, despite HTTP also create TCP socket connection underneath.

Reply:
Join us in the #dev channel on Discord. 

HTTP is a stateless protocol, WebSocket is not and thus has more overhead.

Reply:
I did some more reading on websocket and I think I can see now where the overhead of websocket could be for establishing connection. Establishing websocket connection first require sending/receiving HTTP upgrade request/response before switching to WebSocket connection. Thanks for pointing me out. Learn something new every day.

I have a change that keep the same approach on keeping the connection and retry when ConnectionClosed is raised. This fixes the `turn_off` issue. However, it appears that for `volume_up` and `volume_down` command, there is a separate issue. After some idle time sending these two commands will first result in no response at all. No exception is raised. The subsequent command, however, will raise `BrokenPipeError`. The issue is not a regression from this fix though. To me this appears to be an issue with the underlying library, either `samsungctl` or `websocket-client`. There is a [similar issue](https://github.com/Ape/samsungctl/issues/61) reported in samsungctl.

I can update this PR with the fix for the `turn_off` issue alone and create a separate issue to track the BrokenPipeError issue and tackle that separately (as the latter might require changes in the library). What do you think?

Reply:
It should be fairly simple to have a method `sendCommand`:
 - if closed or never opened, open connection
 - try sending command
 - if we get broken pipe error or other exception, set connection to None and call `sendCommand` again ?

Reply:
Shouldn't that solve the cases?

Reply:
There is also a case where there is no exception raised from connection but TV won't respond. In this case we won't know whether the command succeeded or not. 

I found the issue within samsungctl and working on a way to fix it.

Reply:
Can we then close this PR?

Reply:
Sorry, I forgot that the fixes for the two issues are independent to each other. I'm updating this PR with the fix for turn_off issue. The fix in `samsungctl` for the no response issue can come in later.